src/EFCore/Extensions/EntityFrameworkQueryableExtensions.cs

```
@@ -34,13 +34,14 @@ public static class EntityFrameworkQueryableExtensions
         ///     </para>
         /// </summary>
         /// <param name="source"> The query source. </param>
+        /// <param name="parameterValues"> If not null, then EF will populate the given dictionary with parameter name/value pairs. </param>
         /// <returns> The query string for debugging. </returns>
-        public static string ToQueryString([NotNull] this IQueryable source)
+        public static string ToQueryString([NotNull] this IQueryable source, [CanBeNull] IDictionary<string, object> parameterValues = null)
         {
             Check.NotNull(source, nameof(source));
 
             return source.Provider.Execute<IEnumerable>(source.Expression) is IQueryingEnumerable queryingEnumerable
-                ? queryingEnumerable.ToQueryString()
+                ? queryingEnumerable.ToQueryString(parameterValues)
```

Shouldn't this throw an exception if it's not our IQueryingEnumerable, rather than return this string? Some tool may try to parse this as SQL etc.

---

No. This is like ToString--we want the debugger to show something, even if it is just a generic message.

The idea that people might start parsing the SQL here is really the reason this is controversial. I would consider this abuse of the API, in the same way that parsing ToString output is considered an abuse.